### ISSUE TITLE
iphone 5S - ApNonce does not match APTicket nonce

### ISSUE THREAD

**Issue by jan-kowalski** [LukeZGD/iOS-OTA-Downgrader#270]
Hi,

I'm trying to downgrade iphone 5S to 10.3.3 with linux 22.04.

Script works fine with "trick" to disconnect cable after Gaster PATCH ret:true but at the end I've got message:
`what=Device ApNonce does not match APTicket nonce`

end of log:

```
version: b0a98896ef8d273549f454d45f29985070615996 - 197
img4tool version: 0.197-aca6cf005c94caf135023263cbb5c61a0081804f-RELEASE
Odysseus support: no
INFO: device serial number is DX3S7KGTFFG9
[INFO] 64-bit device detected
futurerestore init done
reading signing ticket 3901167659484_iPhone6,2_n53ap_10.3.3-14G60_c52037e92b177f814b17be06b798072079ff757a.shsh2 is done
[TSSC] opening tmp/BuildManifest.plist
[TSSR] Checking BuildIdentity 0
[TSSR] Selected BuildIdentity for request
[TSSR] User specified to not request a baseband ticket.
Request URL set to https://gs.apple.com/TSS/controller?action=2
Sending TSS request attempt 1... response successfully received
Did set SEP+baseband path and firmware
ERROR: Unable to connect to device?!
ERROR: Unable to get FirmwarePreflightInfo
[WARNING] failed to read BasebandGoldCertID from device! Is it already in recovery?
[WARNING] using tsschecker's fallback to get BasebandGoldCertID. This might result in invalid baseband signing status information
[TSSC] opening tmp/BuildManifest.plist
[TSSR] Checking BuildIdentity 0
[TSSR] Selected BuildIdentity for request
[TSSR] User specified to request only a Baseband ticket.
Request URL set to https://gs.apple.com/TSS/controller?action=2
Sending TSS request attempt 1... response successfully received
Found device in Recovery mode
Device already in recovery mode
Found device in Recovery mode
Identified device as n53ap, iPhone6,2
Extracting BuildManifest from iPSW
Product version: 10.3.3
Product build: 14G60 Major: 14
Device supports Image4: true
Got ApNonce from device: 03 01 14 29 dc a6 e0 e1 e7 1c c9 97 16 f2 7d 94 13 1d 81 28 
Cleaning up...
[exception]:
what=Device ApNonce does not match APTicket nonce

code=44498961
line=679
file=futurerestore.cpp
commit count=197
commit sha  =b0a98896ef8d273549f454d45f29985070615996
Done: restoring failed!
```

Any chance to fix this?

**Comment by gianry** [comment]
Hi,
I've just tried to downgrade an iPad Air ( model 4.2 ) from iOS 12.5.7 to 10.3.3.

I received the same error > 
[exception]:
what=Device ApNonce does not match APTicket nonce


Thanks to all for the support

**Comment by LukeZGD** [comment]
Full log? Also when futurerestore runs, the screen backlight must be on, but it should also be a black screen. If you see the iTunes logo, that means the exploit failed!

**Comment by LukeZGD** [comment]
Fixed in 2023-03-04-d0f19d3
